### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by m-mohr**
**Related Issue(s):** #671, #664, #678, #661, #669 + PR #655

**Proposed Changes:**

This PR addresses numerous changes in a bigger PR as I was to lazy to create 10+ PRs. ;-)

1. Several minor improvements to texts and cleaned-up some formatting, especially tables.
2. Updated the list of extensions related to Collections in the Collection spec.
3. Clarify usage of Commons extension, see #671
4. Tried to stick to my "unwritten" rule that field descriptions should be less than three lines, otherwise put the description below the table.
5. Simplified description for license in Collection and Items. SPDX expressions now disallowed (SPDX License Identifier still allowed), use `various` instead. Also ask for a license link if a SPDX license identifier has been specified.
6. Item field `bbox` refers to GeoJSON bbox spec directly.
7. Link directly to respective GeoJSON RFC chapters.
8. Aligned Asset Definition and Item Assets. #661
9. Fixed Asset Definition example.
10. Merge Catalog Types from STAC Catalogs into front page Design Overview.
11. Added a scope column to the API extensions to make the descriptions for stac_extensions more clear. #669 
12. Explain better how to use stac_extensions. I didn't simply copy/paste the description, but tried to only explain relevant parts where they apply at the moment. #669

**PR Checklist:**

- [ ] This PR has **no** breaking changes. (The only breaking change is the removal of SPDX expressions.)
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-spec/blob/dev/api-spec/README.md#openapi-definitions).

### CODE REVIEW

File: extensions/commons/README.md [hunk 1]
```diff
@@ -4,15 +4,19 @@
 
 A group of STAC Item objects from a single source can share a lot of common metadata. This is especially true with satellite imagery that uses the STAC EO Extension. Rather than including these common metadata fields on every Item, they can be provided into the [STAC Collection](../../collection-spec/README.md) that the STAC Items belong to.
 
+This extensions makes most sense in **static catalogs** to move common information to a central place. This reduces duplication of metadata and makes updates to static catalogs easier. It also makes navigating a static catalog easier as user know in advance which common fields to expect although Items can better be summarized with the `summaries` field in Collections. Therefore it is recommended to use the summaries for this use case and only use the Commons extension to avoid data duplication.
```
Comment (matthewhanson) [inline extensions/commons/README.md]: This extensions -> This extension

Comment (m-mohr) [reply extensions/commons/README.md]: Fixed.

File: extensions/commons/README.md [hunk 2]
```diff
@@ -4,15 +4,19 @@
...
 
+This extension makes most sense in **static catalogs** to move common information to a central place. This reduces duplication of metadata and makes updates to static catalogs easier. It also makes navigating a static catalog easier as user know in advance which common fields to expect although Items can better be summarized with the `summaries` field in Collections. Therefore it is recommended to use the summaries for this use case and only use the Commons extension to avoid data duplication.
+
+In **API implementations** the Commons extension should not be used as it leads to less intuitive search results. For example, a search result for Items may not include the fields a user has searched for as it has been moved to the Collection, which may lead to confusion. Also, data duplication in dynamically created responses is usually not an issue and APIs derived from static catalogs can simply merge the common fields back into the Items for the response. In this case the `stac_extensions` field should not contain the `commons` extension any longer.
```
Comment (cholmes) [inline extensions/commons/README.md]: The other thing to point out here is that API's have the 'fields' parameter, that lets a user get back a smaller payload. As one advantage people sometimes point out in commons is that its smaller size. But fields let the client control the size.

Comment (m-mohr) [reply extensions/commons/README.md]: Added it.

**Pull request merged**